nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/ConsumeTwitter.java

```diff
@@ -338,15 +343,43 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
         session.getProvenanceReporter().receive(flowFile, transitUri);
     }
 
+    @OnPrimaryNodeStateChange
+    public void onPrimaryNodeStateChange(final PrimaryNodeState newState) {
+        if (newState == PrimaryNodeState.PRIMARY_NODE_REVOKED) {
+            stopTweetStreamService(true);
+        }
+    }
+
     @OnStopped
     public void onStopped() {
-        if (tweetStreamService != null) {
-            tweetStreamService.stop();
-        }
-        tweetStreamService = null;
+        stopTweetStreamService(false);
         emptyQueue();
     }
 
+    private synchronized void startTweetStreamService(final ProcessContext context) {
+        if (isInitialized.compareAndSet(false, true)) {
+            tweetStreamService = new TweetStreamService(context, messageQueue, getLogger());
+            tweetStreamService.start();
+        }
+
+    }
+
+    private synchronized void stopTweetStreamService(final boolean printMessageQueueWarning) {
+        if (isInitialized.compareAndSet(true, false)) {
+            if (tweetStreamService != null) {
+                tweetStreamService.stop();
+            }
+            tweetStreamService = null;
+
+            if (printMessageQueueWarning && !messageQueue.isEmpty()) {
+                final String warningMsg = String.format("There are [%s] tweets remaining in the queue, it will only " +
+                        "be processed once this node becomes the primary node again.", messageQueue.size());
+                getLogger().warn(warningMsg);
```

Comment:
`String.format()` should not be used with logging since placeholders can be used and evaluated in the context of the log message. Also recommend adjusting the wording so that the warning is always applicable.
```suggestion
                final String warningMsg = String.format("There are [%s] tweets remaining in the queue, it will only " +
                        "be processed once this node becomes the primary node again.", messageQueue.size());
                getLogger().warn("Stopped consuming stream: unprocessed messages [{}]", messageQueue.size());
```

Reply:
Changing the logger message